proxy/src/rate_limiter/limiter.rs

```diff
@@ -690,4 +688,20 @@ mod tests {
             assert!(limiter.check(endpoint.clone()));
         }
     }
+
+    #[tokio::test]
+    async fn test_rate_limits_gc() {
+        let rand = rand::rngs::StdRng::from_seed([1; 32]);
+        let hasher = BuildHasherDefault::<FxHasher>::default();
```

Comment:
fixed seeded random/hasher to ensure that the test is not flaky

Reply:
And this comment as well.